starfish/imagestack/imagestack.py

```diff
@@ -992,8 +992,16 @@ def max_proj(self, *dims: Indices) -> np.ndarray:
             max projection
 
         """
-        max_projection = self._data.max([dim.value for dim in dims]).values
-        return max_projection
+        max_projection = self._data.max([dim.value for dim in dims])
+        max_projection = max_projection.expand_dims(tuple(dim.value for dim in dims))
+        max_projection = max_projection.transpose(*self.xarray.dims)
+        max_proj_stack = self.from_numpy_array(max_projection.values)
+        return max_proj_stack
+
+    def _squeezed_numpy(self, *dims: Indices):
+        """return this ImageStack's data as a squeezed numpy array"""
+        return xr.DataArray.squeeze(self.xarray,
```

Comment:
I would just do: `self.xarray.squeeze().values`

it's not clear `_squeezed_numpy` needs to limit dims (is there a case where we need it?)

Reply:
If there's an Imagestack that has a dimension of size 1 other then the ones you max project over calling squeeze on the result will accidentally also remove that dimension (because it was already size 1). This came up in the ISS notebook test I think, at one point the imagestack had a shape like (r:1, c:4, z:4, x:40, y:50) but we only did a max projection over ch and z so the resulting stack was (r:1, c:1, z:1, x:40, y:50) and calling squeeze on that incorrectly removed the round dimension. So I added the dims params so you could limit the dimensions to squeeze

Reply:
Ah! That makes sense.

I think you can still do `self.xarray.squeeze((dims...)).values` though.

Reply:
ok.  but why aren't you just calling `self.xarray.squeeze` instead of `xr.DataArray.squeeze(xarray)`